src/algebra/star/basic.lean

```diff
@@ -266,18 +277,32 @@ def star_ring_of_comm {R : Type*} [comm_semiring R] : star_ring R :=
   ..star_monoid_of_comm }
 
 /--
-An ordered `*`-ring is a ring which is both an ordered ring and a `*`-ring,
-and `0 ≤ star r * r` for every `r`.
-
-(In a Banach algebra, the natural ordering is given by the positive cone
-which is the closure of the sums of elements `star r * r`.
-This ordering makes the Banach algebra an ordered `*`-ring.)
+An ordered `*`-ring is a ring which is both an `ordered_add_comm_group` and a `*`-ring,
+and `0 ≤ r ↔ ∃ s, r = star s * s`.
 -/
-class star_ordered_ring (R : Type u) [ordered_semiring R] extends star_ring R :=
-(star_mul_self_nonneg : ∀ r : R, 0 ≤ star r * r)
+class star_ordered_ring (R : Type u) [ring R] [partial_order R] extends star_ring R :=
+(add_le_add_left       : ∀ a b : R, a ≤ b → ∀ c : R, c + a ≤ c + b)
+(nonneg_iff            : ∀ r : R, 0 ≤ r ↔ ∃ s, r = star s * s)
```

Comment:
This seems awfully strong, and means that the rationals are no longer an ordered_star_ring.

Reply:
Yes, it's quite strong, but I don't see any issue with it. For example, the rationals are still a `star_ring` and a `linearly_ordered_field`. But, they shouldn't really be a`star_ordered_ring` because the star structure doesn't determine the order structure.

Reply:
Does this increase in strength correspond to a reduction in generality of `tsirelson_inequality`? Was it previously applicable to `rat`, but after this PR no longer applicable?

@semorrison?

Reply:
Well, the current version of `tsirelson_inequality` doesn't apply to the noncommutative setting (i.e. the one where it is interesting!) because it requires `ordered_semiring`, and with this PR it would. (Also it has square roots of 2 all over the place, so it wouldn't work with `rat` anyway.) I'm also not too sure why `rat` should be a `star_ordered_ring`.

Reply:
Just to add a bit more detail: the goal is to be able to work with C*-algebras with the usual order defined on them. See [here on Wikipedia](https://en.wikipedia.org/wiki/C*-algebra#Structure_of_C*-algebras) for a source that defines it like in this PR; some other sources define it via the spectrum (an element is nonnegative iff its spectrum is real and nonnegative). These definitions are equivalent, but I think this one is more suitable for mathlib since it doesn't involve a second type. I don't see how to get that without the iff.

Reply:
Yes, we definitely don't want to go through the spectrum because then we need to invoke an algebra structure on the ring.

Reply:
Just to confirm, this discussion is resolved, right? @eric-wieser are you happy with things as they are?